test/functional/pdo_sqlsrv/MsCommon.inc

```diff
@@ -52,6 +53,374 @@ function connect($options=array())
+                    }
+                    if ( $noparam ) 
+                    {
+                        $stmt->bindParam( $i, $inputs[$key] );
+                        $i++;
+                    }
+                }
+            }
+            $r = $stmt->execute();
+        }
+        return $stmt;
+    }
+    catch ( Exception $e )
+    {
+        var_dump( $e );
+        exit;
+    }
+}
+
+
+/**
+ * @param string $tbname : name of the table for an insert sql
+ * @param array $input : associative array containing a key value pair of column name and data to put into an insert sql string
+ * @return string a complete insert sql string
+ */
+function get_insertSql_complete( $tbname, $inputs )
+{
+    $colStr = "INSERT INTO $tbname (";
+    $valStr = "VALUES (";
+    foreach( $inputs as $key => $value )
+    {
+        $colStr .= $key . ", ";
+        if ( is_string( $value ))
+            $valStr .= "'" . $value . "', ";
+        else
+            $valStr .= $value . ", ";
+    }
+    $colStr = rtrim( $colStr, ", " ) . ") ";
+    $valStr = rtrim( $valStr, ", " ) . ") ";
+    $insertSql = $colStr . $valStr;
+    return $insertSql;
+}
+
+
+/**
+ * @param string $tbname : name of the table for an insert sql
+ * @param array $input : associative array containing a key value pair of column name and data to put into an insert sql string
+ * @return string an insert sql string with "?" placeholders for all values
+ */
+function get_insertSql_placeholders( $tbname, $inputs )
+{
+    $colStr = "INSERT INTO $tbname (";
+    $valStr = "VALUES (";
+    foreach( $inputs as $key => $value )
+    {
+        $colStr .= $key . ", ";
+    }
+    $colStr = rtrim( $colStr, ", " ) . ") ";
+    $valStr .= get_seq_placeholders( count( $inputs )) . ") ";
+    $insertSql = $colStr . $valStr;
+    return $insertSql;
+}
+
+
+/**
+ * @param string $spname : name of the stored procedure
+ * @param int $num : number of parameters needed for the stored procedure
+ * @return string a call stored procedure sql string with "?" placeholders for all parameters
+ */
+function get_callProcSql_placeholders( $spname, $num )
+{
+    $callStr = "{CALL $spname (";
+    $callStr .= get_seq_placeholders( $num ) . ")} ";
+    return $callStr;
+}
+
+
+/**
+ * @param int $num : number of placeholders needed
+ * @return string a string containing $num number of repeated "?" placeholders delimited by ", "
+ */
+function get_seq_placeholders( $num )
+{
+    $placeholderStr = str_repeat( "?, ", $num );
+    $placeholderStr = rtrim( $placeholderStr, ", " );
+    return $placeholderStr;
+}
+
+
+/**
+ * Fetch all rows and all columns given a table name, and print them
+ * @param object $conn : PDO connection object
+ * @param string $tbname : name of the table to fetch from
+ */
+function fetch_all( $conn, $tbname ) {
+    try
+    {
+        $sql = "SELECT * FROM $tbname";
+        $stmt = $conn->query( $sql );
+        while ( $row = $stmt->fetch( PDO::FETCH_ASSOC ))
+        {
```

Comment:
perhaps the fetch type is not hardcoded but defaulted to FETCH_ASSOC?

Reply:
This function prints the result set and the user does not get the result set back. I don't think a user would want to see a numeric key being printed ( e.g., "1: blah" ) since it's very uninformative.

Reply:
For your current test it makes sense to use FETCH_ASSOC but since this method is a COMMON method for other tests to use, it will be nice to make it more flexible

Reply:
The function doesn't return the result set though, they just get printed.. most user prefer numeric keys for convenience, but in this case the user doesn't have access to the result set being fetched. I still don't see when a user would prefer to see numeric keys being printed.

A common function can be created in the future that returns the result set with optional fetch types.